test/unit/app/utils/test_utils_user.py

```diff
@@ -140,3 +148,29 @@ def test_send_welcome_sms(mocker, test_client, init_database, mock_sms_apis,
     messages = mock_sms_apis
     assert messages == [{'phone': '+61123456789', 'message': expected_welcome},
                         {'phone': '+61123456789', 'message': expected_terms},]
+
+def test_attach_transfer_account_to_user(test_client, init_database, create_organisation):
+    with db.session.no_autoflush:
+        organisation = create_organisation
+        organisation.external_auth_password = config.EXTERNAL_AUTH_PASSWORD
+
+        g.active_organisation = organisation
+
+        user = UserFactory(id=20,
+                           phone='+61756432178',
+                           first_name='Unknown first name',
+                           last_name='Unknown last name',
+                           registration_method=RegistrationMethodEnum.USSD_SIGNUP)
+
+        user.add_user_to_organisation(organisation, False)
+
+        assert user.primary_blockchain_address is None
+
+        user_with_transfer_account = attach_transfer_account_to_user(user)
+
+        assert user_with_transfer_account.primary_blockchain_address is not None
+        bio = next(filter(lambda x: x.name == 'bio', user_with_transfer_account.custom_attributes), None)
```

Comment:
are the custom attributes linked to the transfer account? if not, let's remove the checks that aren't in context.